seismicpro/src/refractor_velocity/refractor_velocity.py

```diff
@@ -0,0 +1,574 @@
+"""Implements RefractorVelocity class for estimating the velocity model of an upper part of the section."""
+
+from functools import partial
+
+import numpy as np
+from sklearn.linear_model import SGDRegressor
+from scipy import optimize
+
+from ..decorators import plotter
+from ..utils import set_ticks, set_text_formatting
+from ..utils.interpolation import interp1d
+
+
+# pylint: disable=too-many-instance-attributes, protected-access
+class RefractorVelocity:
+    """The class stores and fits parameters of a velocity model of an upper part of the section.
+
+    An instance can be created using one of the following `classmethod`s:
+        * `from_first_breaks` - fits a near-surface velocity model by offsets and times of first breaks.
+        * `from_params` - creates a `RefractorVelocity` instance from given parameters without model fitting.
+        * `from_constant` - creates a single-layer `RefractorVelocity` with zero intercept time and given velocity of
+                            the refractor.
+
+    Some of class attributes and classmethod's arguments have a valid key notation.
+        `t0`: an intercept time of the first subweathering layer. Same as two-way travel time to the first
+              refractor. Measured in milliseconds.
+        `x{i}`: cross offset. The offset where refracted wave from the i-th layer comes at the same time with
+                a refracted wave from the next underlaying layer. Measured in meters.
+        `v{i}`: velocity of the i-th layer. Measured in meters per seconds.
+
+    The RefractorVelocity created with `from_first_breaks` classmethod calculates the missing parameters from passed
+    parameters. The passed `init`, `bounds` and `n_refractors` parameters combine with the calculated parameters and
+    stored in class atributes with corresponding names. All passed parameters have a greater priority.
```

Comment:
```suggestion
    The same names are used as keys in `init` and `bounds` dicts passed to `RefractorVelocity.from_first_breaks` constructor. Some keys may be omitted in one dict if it's passed in another, e.g. one can pass only bounds for `v1` without an initial value, which will be inferred automatically. Both `init` and `bounds` dicts may not be passed at all if `n_refractors` is given.

```
And remove all "valid key notation", this line + all the examples below should be enough